src/cargo/core/compiler/layout.rs

```diff
@@ -170,6 +172,7 @@ impl Layout {
             fingerprint: dest.join(".fingerprint"),
             examples: dest.join("examples"),
             doc: root.join("doc"),
+            temp_root: Self::temp_root_path(&root),
```

Comment:
The naming is a bit confusing, because cache and temp are usually different concepts of which people have different conceptions.